registry/handlers/hooks.go

```diff
@@ -0,0 +1,103 @@
+package handlers
+
+import (
+	"fmt"
+	"net/smtp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/configuration"
+)
+
+// ReportHook is for hooking Panic in web application
+type ReportHook struct {
+}
+
+// Fire forwards an error to ReportHook
+func (hook *ReportHook) Fire(entry *logrus.Entry) error {
+	logrus.Info("Message: ", entry.Message)
+	if errConfig.Log.Handlers.Errors.Type == "email" {
+		if err := mailSending(errConfig, entry.Message); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Levels contains hook levels to be catched
+func (hook *ReportHook) Levels() []logrus.Level {
+	return []logrus.Level{
+		logrus.DebugLevel,
+		logrus.InfoLevel,
+		logrus.WarnLevel,
+		logrus.ErrorLevel,
+		logrus.FatalLevel,
+		logrus.PanicLevel,
+	}
+}
+
+var reportLevel = map[string]logrus.Level{
+	"debug": logrus.DebugLevel,
+	"info":  logrus.InfoLevel,
+	"warn":  logrus.WarnLevel,
+	"error": logrus.ErrorLevel,
+	"fatal": logrus.FatalLevel,
+	"panic": logrus.PanicLevel,
+}
+
+var errConfig = &configuration.Configuration{}
+
+// ErrorReport is for using message with hook, ex. ErrorReport.Panic
+var ErrorReport = logrus.New()
```

Comment:
This creates a new logger, which is not the right thing. Use `GetLogger(ctx)` to get the current logger and push the configured result using `WithLogger(ctx, logger)`.